modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1214 @@
+
+After some improvements to the `fontsizeinput` toolbar item, pressing enter in the input field now sets the focus back to the {productname} editor viewport, which allows the new font size to be used. 
+
+In {productname} 6.4.1, pressing enter inside the `fontsizeinput` input now returns the focus back to the editor content body.
+
+
+[[additions]]
+== Additions
+
+{productname} 6.4.1 also includes the following additions:
+
+=== New tree component that can be used in dialog body panel
+//#TINY-9532
+
+A new tree component has been added to {productname}, that can be used in custom plugins and models.
+
+The new tree component is designed to create a visual representation of the hierarchical structure of items such as, showcasing folders and the files within these folders, similar to how a directory works which allows users to easily identify various structure levels within set categories, such as how the new `advtemplate` works.
+
+This new tree component is an important addition to {productname} which will be used in several features with up and coming releases in the future.
+
+=== renderUI property in the Theme type can now return a Promise <RenderResult> instead of RenderResult
+//#TINY-9556
+
+In earlier versions of {productname} the `renderUI` function which initiates theme rendering did not wait for the CSS to load. This could cause race conditions with other parts of the editor or customer integrations. As a result {productname} 6.4.1 now waits for a CSS load event before returning from the `renderUI` function. This process necessitates changing the return type of the `renderUI` function; it now returns a `Promise<RenderResult>`. The type is `Promise<RenderResult> | RenderResult` for compatibility, but the `silver` theme always returns a promise.
+
+This change should have no impact on customer integrations, even those implementing custom themes.
+
+=== New `editor.selection.isEditable` API method that returns whether the current selection is editable
+//#TINY-9462
+
+In previous versions of {productname}, when the body was set to `contenteditable="false"` content could still be inserted with the `insertContent` API.
+
+Example content:
+[source, HTML]
+----
+<body contenteditable="false"><p>Try to insert content here</p></body>
+----
+
+After selecting the space between "insert" and "content", executing `tinymce.activeEditor.insertContent('text')` would incorrectly insert `text` at the selection which should not be possible.
+
+[source, HTML]
+----
+<body contenteditable="false"><p>Try to inserttextcontent here</p></body>
+----
+
+{productname} 6.4 includes a new `isEditable` method on both the `editor.selection` and `editor.dom` APIs which are used to block the `editor.insertContent` API and `mceInsertContent` command when the selection covers a `contenteditable="false"` element.
```

Comment:
Since the insertContent blocking is described in it's own section later it might be better to just describe the API here what that does. Hard to describe exactly what it's doing but I gave it a shot here.

```suggestion
=== New `editor.selection.isEditable` API method that returns whether the current selection is editable
//#TINY-9462

{productname} 6.4 includes a new `isEditable` method on the `editor.selection` that will check if the selection  is editable by finding the closest ascendant element for each selection endpoint with either a `true` or `false` value for it's `contenteditable` attribute. If both endpoints of the selection has it's closest ancestor `contenteditable` attribute value of `true` then the selection is editable.
```